src/DynamoCoreWpf/HelixRenderPackage.cs

```diff
@@ -0,0 +1,560 @@
+        {
+            get { return mesh.Normals.ToDoubleArray(); }
+        }
+
+        public IEnumerable<double> MeshTextureCoordinates
+        {
+            get { return mesh.TextureCoordinates.ToDoubleArray(); }
+        }
+
+        public IEnumerable<double> PointVertices
+        {
+            get { return points.Positions.ToDoubleArray(); }
+        }
+
+        public IEnumerable<byte> PointVertexColors
+        {
+            get { return points.Colors.ToByteArray(); }
+        }
+
+        public IEnumerable<int> PointIndices
+        {
+            get { return points.Indices.ToArray(); }
+        }
+
+        #endregion
+
+        public MeshGeometry3D Mesh
+        {
+            get { return mesh; }
+        }
+
+        public LineGeometry3D Lines
+        {
+            get { return lines; }
+        }
+
+        public PointGeometry3D Points
+        {
+            get
+            {
+                return points;
+            }
+        }
+
+        internal static LineGeometry3D InitLineGeometry()
+        {
+            var lines = new LineGeometry3D
+            {
+                Positions = new Vector3Collection(),
+                Indices = new IntCollection(),
+                Colors = new Color4Collection()
+            };
+            
+            return lines;
+        }
+
+        internal static PointGeometry3D InitPointGeometry()
+        {
+            var points = new PointGeometry3D()
+            {
+                Positions = new Vector3Collection(),
+                Indices = new IntCollection(),
+                Colors = new Color4Collection()
+            };
+
+            return points;
+        }
+
+        internal static MeshGeometry3D InitMeshGeometry()
+        {
+            var mesh = new MeshGeometry3D()
+            {
+                Positions = new Vector3Collection(),
+                Indices = new IntCollection(),
+                Colors = new Color4Collection(),
+                Normals = new Vector3Collection(),
+                TextureCoordinates = new Vector2Collection()
+            };
+
+            return mesh;
+        }
+
+        internal static BillboardText3D InitText3D()
+        {
+            var text3D = new BillboardText3D();
+
+            return text3D;
+        }
+
+        internal static string CleanTag(string tag)
+        {
+            var splits = tag.Split(':');
+            return splits.Count() <= 1 ? "[0]" : string.Format("[{0}]",string.Join(",", splits.Skip(1)));
+        }
+
+        private static Color4 Color4FromBytes(byte red, byte green, byte blue, byte alpha)
+        {
+            var color = new Color4((red / 255.0f),(green / 255.0f),(blue / 255.0f), (alpha/255.0f));
+            return color;
+        }
+
+        private static Vector3 Vector3ForYUp(double x, double y, double z)
+        {
+            return new Vector3((float)x, (float)z, (float)-y);
+        } 
+    }
+
+    internal static class HelixRenderExtensions
+    {
```

Comment:
None of the below methods return an array.  Consider changing to something more accurate.  Like, `ToDoubleEnumerable`  or `ToEnumerable`